var/spack/repos/builtin/packages/tcsh/package.py

```diff
@@ -15,9 +15,11 @@ class Tcsh(AutotoolsPackage):
     syntax."""
 
     homepage = "http://www.tcsh.org/"
-    url      = "ftp://ftp.astron.com/pub/tcsh/tcsh-6.20.00.tar.gz"
+    url      = "http://ftp.funet.fi/pub/mirrors/ftp.astron.com/pub/tcsh/old/tcsh-6.20.00.tar.gz"
 
-    version('6.20.00', '59d40ef40a68e790d95e182069431834')
+    version('6.21.00', sha256='c438325448371f59b12a4c93bfd3f6982e6f79f8c5aef4bc83aac8f62766e972',
+            url='http://ftp.funet.fi/pub/mirrors/ftp.astron.com/pub/tcsh/tcsh-6.21.00.tar.gz')
+    version('6.20.00', sha256='59d40ef40a68e790d95e182069431834')
```

Comment:
This is an MD5 checksum, not SHA256.  See below for the correct SHA256 checksum:
```
$ curl -O http://ftp.funet.fi/pub/mirrors/ftp.astron.com/pub/tcsh/old/tcsh-6.20.00.tar.gz

$ md5 tcsh-6.20.00.tar.gz 
MD5 (tcsh-6.20.00.tar.gz) = 59d40ef40a68e790d95e182069431834

$ shasum -a 256 tcsh-6.20.00.tar.gz 
b89de7064ab54dac454a266cfe5d8bf66940cb5ed048d0c30674ea62e7ecef9d
```

Reply:
Thank you for your check.
I changed a checksum for 6.20.00 to an SHA256 checksum.